tracing-attributes/src/expand.rs

```diff
@@ -49,8 +50,35 @@ pub(crate) fn gen_function<'a, B: ToTokens + 'a>(
 
     let warnings = args.warnings();
 
+    let block = if let ReturnType::Type(_, return_type) = &output {
+        let return_type = erase_impl_trait(return_type);
+        // Install a fake return statement as the first thing in the function
+        // body, so that we eagerly infer that the return type is what we
+        // declared in the async fn signature.
+        let fake_return_edge = quote_spanned! {return_type.span()=>
+            #[allow(unreachable_code)]
+            if false {
```

Comment:
take it or leave it, not a big deal: maybe it's worth putting the comment _inside_ the quoted code, so that the comment is expanded in `cargo expand` output? the `if false` block is kind of weird if you don't know what it's for, so maybe it's worth including the comment in case anyone's looking at the expanded code, so that they know why we generate this?

Reply:
ah, well. nevermind then!